src/Symfony/Bundle/SecurityBundle/Resources/config/security_listeners.php

```diff
@@ -32,19 +31,14 @@
 return static function (ContainerConfigurator $container) {
     $container->services()
 
-        ->set('security.authentication.retry_entry_point', RetryAuthenticationEntryPoint::class)
```

Comment:
The service must be deprecated indeed.

>  maybe even add a compiler pass to fix ChannelListener

What would that compiler pass do? ChannelListener can use the legacy $entryPoint when it's injected, right?

Reply:
> What would that compiler pass do? ChannelListener can use the legacy $entryPoint when it's injected, right?

The entry point service must only be injected if a user changed it (otherwise, an unfixable deprecation will be triggered).

Reply:
we don't go that far AFAIK, the `ChannelListener` does not need to keep the deprecated service as dependency